server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -86,14 +86,19 @@ private static String toLoggerName(final Class<?> cls) {
      * Logs a message at the {@link #DEPRECATION} level. The message is also sent to the header warning logger,
      * so that it can be returned to the client.
      */
-    public DeprecationLoggerBuilder deprecate(final String key, final String msg, final Object... params) {
-        return new DeprecationLoggerBuilder().withDeprecation(key, msg, params);
+    public DeprecationLoggerBuilder deprecate(
```

Comment:
btw do you think it would be worthy to make this method work in a "fluent" way?
I initialliy thought that deprecation logger could be used like
```
deprecationLogger.deprecate("msg")
.compatibleWarning("msg2")
```
The compatible logger is still not merged (going to be soon I hope), but maybe we could use the builder here?
I am worried about the performance and we won't be creating too much objects here though..
WDYT?

Reply:
I don't follow - `deprecationLogger.deprecate(...)` already returns a `DeprecationLoggerBuilder`, so isn't it already fluent?